applications/swimming_DEM_application/custom_elements/spheric_swimming_particle.cpp

```diff
@@ -701,10 +720,31 @@ void SphericSwimmingParticle<TBaseElement>::ComputeBassetForce(NodeType& node, a
             }
 
             array_1d<double, 3> basset_term    = fractional_derivative_of_slip_vel;
+            array_1d<double, 3> old_basset_term;
             const array_1d<double, 3>& vel     = node.FastGetSolutionStepValue(VELOCITY);
             const array_1d<double, 3>& old_vel = node.FastGetSolutionStepValue(VELOCITY_OLD);
-            fractional_derivative_of_slip_vel -= mOldBassetTerm + mOldDaitchePresentCoefficient * (old_vel - vel); // the second term corresponds to the part that was treated implicitly in the last step minus a part that was added but did not correspond to the basset term
-            mOldBassetTerm = basset_term;
+            old_basset_term = mOldBassetTerm + mOldDaitchePresentCoefficient * (old_vel - vel); // the second term corresponds to the part that was treated implicitly in the last step minus a part that was added but did not correspond to the basset term
+            noalias(mOldBassetTerm) = basset_term;
+
+            if (r_current_process_info[FRAME_OF_REFERENCE_TYPE] >= 1){
+                array_1d<double, 3>& displacement_old = node.FastGetSolutionStepValue(DISPLACEMENT_OLD);
+                array_1d<double, 3>& displacement_old_old = node.FastGetSolutionStepValue(VELOCITY_OLD_OLD);
+                array_1d<double, 3> aux = mOldDaitchePresentCoefficient * (displacement_old_old - displacement_old);
```

Comment:
I am not 100% sure, but I think 
```
array_1d<double, 3> aux;
noalias(aux)=mOldDaitchePresentCoefficient * (displacement_old_old - displacement_old);
```
could be faster. Maybe @RiccardoRossi can confirm this.

Reply:
hard to say. It depends on how ublas does it internally. IF it was well optimized it would be the same, but right now i cannot say...

Riccardo